### PR TITLE
allow users to set partitioner

### PR DESCRIPTION
**Pull request by ksagiyam**
This PR is to:
- define `set_partitioner` method in `MeshTopology` class
- allow users to set `partitioner` as:
`mesh = Mesh(..., distribution_parameters={'partitioner_type': 'parmetis'})`

### CODE REVIEW

File: firedrake/mesh.py [hunk 1]
```diff
@@ -922,6 +910,52 @@ def cell_set(self):
         size = list(self._entity_classes[self.cell_dimension(), :])
         return op2.Set(size, "Cells", comm=self.comm)
 
+    def set_partitioner(self, distribute, partitioner_type=None):
+        """Set partitioner for (re)distributing underlying plex over comm.
+
+        :arg distribute: Boolean or (sizes, points)-tuple.  If (sizes, point)-
+            tuple is given, it is used to set shell partition. If Boolean, no-op.
+        :kwarg partitioner_type: Partitioner to be used: "chaco", "ptscotch", "parmetis",
+            "shell", or `None` (unspecified).
```
Comment (wence-) [inline firedrake/mesh.py:919]: "shell" does not seem to be a valid type from code below.

Comment (wence-) [reply firedrake/mesh.py:919]: No, I see.

Comment (wence-) [inline firedrake/mesh.py:919]: Say in docstring that `partitioner_type` is ignored if the distribute parameter specifies the distribution.

**Review by wence- — COMMENTED** [review]
Minor comments on docstring, otherwise 👍

File: firedrake/mesh.py (hunk 1 above)
Comment (ksagiyam) [reply firedrake/mesh.py]: Done. Thanks!

**Pull request merged**